api/proquest/importer.py

```diff
@@ -390,6 +389,49 @@ def _get_book(self, patron, configuration, document_id):
 
             iterations += 1
 
+    def _extract_image_links(self, publication, feed_self_url):
+        """Extracts a list of LinkData objects containing information about artwork.
+
+        :param publication: Publication object
+        :type publication: ast_core.Publication
+
+        :param feed_self_url: Feed's self URL
+        :type feed_self_url: str
+
+        :return: List of links metadata
+        :rtype: List[LinkData]
+        """
+        self._logger.debug(
+            u"Started extracting image links from {0}".format(
+                encode(publication.images)
+            )
+        )
+
+        image_links = []
+
+        for image_link in publication.images.links:
+            cover_link = self._extract_link(
+                image_link,
+                feed_self_url,
+                default_link_rel=Hyperlink.IMAGE,
+            )
+            image_links.append(cover_link)
+
+            thumbnail_link = self._extract_link(
+                image_link,
+                feed_self_url,
+                default_link_rel=Hyperlink.THUMBNAIL_IMAGE,
+            )
+            image_links.append(thumbnail_link)
```

Comment:
cover_link will be a `LinkData`, and `LinkData` has a `.thumbnail` attribute, so you should set `cover_link.thumbnail = thumbnail_link` rather than adding `thumbnail_link` to `image_links`. Otherwise it's not explicit that `thumbnail_link` is a thumbnail _of_ `cover_link` -- it could be a thumbnail of some other image.

metadata_layer might be smart enough to figure it out, since there's one image and one thumbnail, but it's better to be explicit.

Reply:
I see how it's working now, because the Proquest feed doesn't specify a link relation for its cover image, but  `default_link_rel` won't do anything if a link relation is specified in the OPDS. In particular, if Proquest starts specifying rel="image", you'll end up with the thumbnail having a rel="image", which metadata_layer will reject.

Since you know these are images (they're in the `images` section of the OPDS entry), i would explicitly set `cover_link.rel` and `thumbnail_link.rel` rather than (or in addition to) expecting `default_link_rel` to do it.